[BE] delete spmd module

wanchaol opened the pull request:
Summary:
as titled, fully delete spmd module as we stopped working on this and the code is already broken with no unit tests enabled.

We should not keep it in the codebase as it provide no value anymore, and it burdens DTensor to maintain the compatiblity with it (i.e. code paths/imports) constantly.

Test Plan: sandcastle

Differential Revision: D60402105




cc @XilunWu @H-Huang @awgu @kwen2501 @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o @albanD

albanD reviewed (approved):
Thanks for the cleanup!

Commit:
[BE] delete spmd module (#132072)

Summary:

as titled, fully delete spmd module as we stopped working on this and the code is already broken with no unit tests enabled.

We should not keep it in the codebase as it provide no value anymore, and it burdens DTensor to maintain the compatiblity with it (i.e. code paths/imports) constantly.

Test Plan: sandcastle

Reviewed By: albanD, fegin, seemethere, awgu

Differential Revision: D60402105
